near-sdk/src/environment/env.rs

```diff
@@ -506,10 +506,20 @@ pub fn panic(message: &[u8]) -> ! {
     unsafe { sys::panic_utf8(message.len() as _, message.as_ptr() as _) }
     unreachable!()
 }
+/// Logs the string message message. This message is stored on chain.
+pub fn log_str(message: &str) {
```

Comment:
Should it be accepting anything that implements `ToString` trait? WDYT @matklad ?
Btw, is this method going to support no-std?

Reply:
I don't see a huge benefit in supporting this because it would increase code size for the monomorphization of the generic and for more general usage, `log!` macro is used. I originally made it generic based on `AsRef<str>` but since it added code size (tested and an example given above) the macro calls `as_ref` and this base call remains as a concrete type. Having it be `ToString` would force an unnecessary allocation that I'm not sure can be optimized away by the compiler.

As for no_std, yes it is no_std compatible and has the same memory layout as bytes. It's currently used as such in the no_std experimental SDK I've been playing with.

Reply:
No, `ToString` would force allocation, and won't be enough to make a nice API. For the latter, we need a format-style macro.